src/haz3lcore/tiles/Base.re

```diff
@@ -13,28 +13,28 @@ type kind =
   | TextArea;
 
 [@deriving (show({with_path: false}), sexp, yojson)]
-type segment = list(piece)
-and piece =
-  | Tile(tile)
+type segment('a) = list(piece('a))
+and piece('a) =
+  | Tile(tile('a))
   | Grout(Grout.t)
   | Secondary(Secondary.t)
-  | Projector(projector)
-and tile = {
+  | Projector(projector('a))
+and tile('a) = {
   // invariants:
   // - length(mold.in_) + 1 == length(label)
   // - length(shards) <= length(label)
   // - length(shards) == length(children) + 1
   // - sort(shards) == shards
-  id: Id.t,
+  extra: 'a,
```

Comment:
Rename this to something more relevant and have the inhabitants use a `{id: Id.t}` type rather than directly embed Id.t